src/simulator/views/gui/view_editor.py

```diff
@@ -620,153 +712,66 @@ def __init__(self, services: Services, voxel_map: VoxelMap):
-                       'OBSTACLES_SQR_WF': [1.0, 1.0, 1.0, 0.5], 'TRAVERSABLES': [1.0, 1.0, 1.0, 0.5],
-                       'TRAVERSABLES_TRI_WF': [1.0, 1.0, 1.0, 0.5], 'TRAVERSABLES_SQR_WF': [1.0, 1.0, 1.0, 0.5],
-                       'START': [0.5, 0.0, 0.5, 1.0], 'GOAL': [0.0, 1.0, 0.0, 0.5], 'PATH': [1.0, 1.0, 1.0, 0.5],
-                       'FRINGE': [0.2, 0.2, 0.8, 1.0], 'EXPLORED': [0.4, 0.4, 0.4, 1.0]},
-
-                      {'OBSTACLES': [0.8, 0.2, 0.2, 1.0], 'OBSTACLES_TRI_WF': [1.0, 1.0, 1.0, 1.0],
-                       'OBSTACLES_SQR_WF': [1.0, 1.0, 1.0, 0.5], 'TRAVERSABLES': [1.0, 1.0, 1.0, 0.5],
-                       'TRAVERSABLES_TRI_WF': [1.0, 1.0, 1.0, 0.5], 'TRAVERSABLES_SQR_WF': [1.0, 1.0, 1.0, 0.5],
-                       'START': [0.5, 0.0, 0.5, 1.0], 'GOAL': [0.0, 1.0, 0.0, 0.5], 'PATH': [1.0, 1.0, 1.0, 0.5],
-                       'FRINGE': [0.2, 0.2, 0.8, 1.0], 'EXPLORED': [0.4, 0.4, 0.4, 1.0]},
-
-                      {'OBSTACLES': [0.8, 0.2, 0.2, 1.0], 'OBSTACLES_TRI_WF': [1.0, 1.0, 1.0, 1.0],
-                       'OBSTACLES_SQR_WF': [1.0, 1.0, 1.0, 0.5], 'TRAVERSABLES': [1.0, 1.0, 1.0, 0.5],
-                       'TRAVERSABLES_TRI_WF': [1.0, 1.0, 1.0, 0.5], 'TRAVERSABLES_SQR_WF': [1.0, 1.0, 1.0, 0.5],
-                       'START': [0.5, 0.0, 0.5, 1.0], 'GOAL': [0.0, 1.0, 0.0, 0.5], 'PATH': [1.0, 1.0, 1.0, 0.5],
-                       'FRINGE': [0.2, 0.2, 0.8, 1.0], 'EXPLORED': [0.4, 0.4, 0.4, 1.0]}
-                      ]
-
-        if os.path.isfile('state.json'):
-            with open('state.json', 'r') as json_file:
-                self.views = json.load(json_file)
-                # todo: actually check that structure is the same
+            self.__view_selectors.append(DirectButton(image=num,
+                                                      pos=(-0.7 + i * 0.7, 0.4, -4.9),
+                                                      parent=self.__window.frame,
+                                                      scale=0.16,
+                                                      frameColor=TRANSPARENT,
+                                                      command=lambda v: setattr(self, "view_idx", v),
+                                                      extraArgs=[i+3]))
+
+        self.__elems = []
+        self.__reset()
+
+    def __reset(self) -> None:
+        self.__services.state.restore_effective_view()
+        ps_colours = self.__services.state.effective_view.colours
+
+        for e in self.__elems:
+            e.destroy()
+        self.__elems.clear()
+
+        for name, dc in ps_colours.items():
+            self.__elems.append(ViewElement(name, self.__update_visibility, self.__update_colour, self.__select_cv, self.__window.frame, dc.visible, dc.colour))
+
+        for i in range(len(self.__elems)):
+            self.__elems[i].frame.set_pos((0.0, 0.0, -1.9 - 0.2 * i))
+
+        if self.view_idx < 3:
```

Comment:
Can use the same calculation as above for this `if` statement.

Reply:
Done